qa/rpc-tests/mempool_push.py

```diff
@@ -0,0 +1,163 @@
+        print("Initializing test directory "+self.options.tmpdir)
+        # pick this one to start from the cached 4 node 100 blocks mined configuration
+
+        initialize_chain(self.options.tmpdir, bitcoinConfDict, wallets)
+        # pick this one to start at 0 mined blocks
+        # initialize_chain_clean(self.options.tmpdir, 4, bitcoinConfDict, wallets)
+        # Number of nodes to initialize ----------> ^
+
+    def setup_network(self, split=False):
+        mempoolConf = [
+            ["-blockprioritysize=2000000"],
+            ["-blockprioritysize=2000000", "-maxmempool=8080", "-limitancestorsize=%d" % (BCH_UNCONF_SIZE_KB*2), "-limitdescendantsize=%d" % (BCH_UNCONF_SIZE_KB*2), "-limitancestorcount=%d" % (BCH_UNCONF_DEPTH*2), "-limitdescendantcount=%d" % (BCH_UNCONF_DEPTH*2), "-net.unconfChainResendAction=2"],
+            ["-blockprioritysize=2000000", "-limitdescendantcount=1000", "-limitancestorcount=1000","-limitancestorsize=1000", "-limitdescendantsize=1000", "-net.unconfChainResendAction=2"],
+            ["-blockprioritysize=2000000", "-limitancestorsize=150","-net.unconfChainResendAction=2"]
+            ]
+        self.nodes = start_nodes(4, self.options.tmpdir, mempoolConf)
+        connect_nodes_full(self.nodes)
+        self.is_network_split=False
+        self.sync_blocks()
+
+    def run_test (self):
+        # kick us out of IBD mode since the cached blocks will be old time so it'll look like our blockchain isn't up to date
+        # if we are in IBD mode, we don't request incoming tx.
+        self.nodes[0].generate(1)
+
+        logging.info("ancestor count test")
+        bal = self.nodes[1].getbalance()
+        addr = self.nodes[1].getnewaddress()
+
+        txhex = []
+        for i in range(0,BCH_UNCONF_DEPTH*2):
+          try:
+            txhex.append(self.nodes[1].sendtoaddress(addr, bal-1))  # enough so that it uses all UTXO, but has fee left over
+          except JSONRPCException as e: # an exception you don't catch is a testing error
+              print(str(e))
+              raise
+        assert_equal(self.nodes[0].getmempoolinfo()["size"], BCH_UNCONF_DEPTH)
+        assert_equal(self.nodes[1].getmempoolinfo()["size"], BCH_UNCONF_DEPTH*2)
+        # Set small to commit just a few tx so we can see if the missing ones get pushed
+        self.nodes[0].set("mining.blockSize=6000")
+        blk = self.nodes[0].generate(1)[0]
+        blkhex = self.nodes[0].getblock(blk)
+        waitFor(DELAY_TIME, lambda: self.nodes[0].getmempoolinfo()["size"] == BCH_UNCONF_DEPTH)
+        # generate the block somewhere else and see if the tx get pushed
+        self.nodes[2].set("mining.blockSize=4000")
+        blk2 = self.nodes[2].generate(1)[0]
+        waitFor(DELAY_TIME, lambda: self.nodes[0].getmempoolinfo()["size"] == BCH_UNCONF_DEPTH)
+
+        waitFor(DELAY_TIME, lambda: self.nodes[1].getbestblockhash() == blk2)  # make sure its settled so we can get a good leftover count for the next test.
```

Comment:
format nit: I find difficult to review with inline comment that makes line so long